HeroGame/hero.py

```diff
@@ -0,0 +1,101 @@
+import math
+
+
+class Hero:
+    heal_percent = 0.20  # 10%
+    level_up_benefits_percent = 0.50  # 20%
+    level_up_threshold = 3
+
+    def __init__(self, name):
+        self._name = name
+        self._level = 1
+        self._damage = 2
+        self._hp = 10
+        self._max_health = 10
+        self._coins = 0
+        self.defending = False
+
+    @property
+    def defending(self) -> bool:
+        return self._defending
+
+    @defending.setter
+    def defending(self, is_defending: bool) -> None:
+        self._defending = is_defending
+
+    def heal(self) -> None:
+        healing = math.ceil(self.hp * Hero.heal_percent)
+        if healing > self.max_health:
+            self.hp = self.max_health
+        else:
+            self.hp += healing
```

Comment:
here you are adding not updating, so if you dont want to pass a max_health you might pass it